runtime/src/main/scala/akka/grpc/internal/GrpcResponseHelpers.scala

```diff
@@ -39,6 +39,22 @@ object GrpcResponseHelpers {
       system: ClassicActorSystemProvider): HttpResponse =
     GrpcResponseHelpers(e, Source.single(GrpcEntityHelpers.trailer(Status.OK)), eHandler)
 
+  def responseForSingleElement[T](e: T, eHandler: ActorSystem => PartialFunction[Throwable, Trailers])(
+      implicit m: ProtobufSerializer[T],
+      writer: GrpcProtocolWriter,
+      system: ClassicActorSystemProvider): HttpResponse = {
+    def x = eHandler
+    def f = system
+    if (false) { x; f }
+
+    val data = DataFrame(m.serialize(e))
```

Comment:
Since `data` is an intermediary representation before encodeFrame, I'd probably pass it directly into encodeFrame.